crates/ide-db/src/imports/import_assets.rs

```diff
@@ -536,6 +537,32 @@ fn trait_applicable_items(
         required_assoc_items.insert(assoc);
         Some(assoc_item_trait.into())
     })
+    .filter(|candidate_trait_id| {
+        // blanket `Trait` implementations for type `A` can only exist in the crate defining
+        // the trait. However, an implementation for `A`` can only exist in `A`'s or `Trait`'s crate,
+        // which allows us to reduce the search space substantially.
+        let candidate_trait_id: TraitId = *candidate_trait_id;
+        let definining_crate_for_trait = db.trait_environment(candidate_trait_id.into()).krate;
+
+        let Some(receiver_fingerprint) =
+            TyFingerprint::for_trait_impl(&trait_candidate.receiver_ty.ty)
+        else {
+            return false;
+        };
+        let definitions_exist_in_trait_crate = db
+            .trait_impls_in_crate(definining_crate_for_trait)
+            .for_trait_and_self_ty(candidate_trait_id, receiver_fingerprint)
```

Comment:
Might be nice to add a method `has_impls_for_trait_and_self_ty(..) -> bool`

Reply:
done!